mvpa2/algorithms/benchmarks/hyperalignment.py

```diff
@@ -0,0 +1,129 @@
+        overlapping_windows=True,
+        distance='correlation',
+        do_zscore=True,
+        clf_direction_correct_way=True):
+    """Time-segment classification across subjects using Hyperalignment
+    """
+    # Generate outer-most partitioning ()
+    parts = [copy.deepcopy(part1).generate(ds) for ds in dss]
+
+    iter = 1
+    errors = []
+    while True:
+        try:
+            dss_partitioned = [p.next() for p in parts]
+        except StopIteration:
+            # we are done -- no more partitions
+            break
+        print "iteration ", iter
+        dss_train, dss_test = zip(*[list(Splitter("partitions").generate(ds))
+                                    for ds in dss_partitioned])
+
+        # TODO:  allow for doing feature selection
+        # Now let's do hyperalignment
+        hyper = copy.deepcopy(hyper)
+        if do_zscore:
+            for ds in dss_train + dss_test:
+                zscore(ds, chunks_attr=None)
+
+        mappers = hyper(dss_train)
+
+        # dss_train_aligned = [mapper.forward(ds) for mapper, ds in zip(mappers, dss_train)]
+        dss_test_aligned = [mapper.forward(ds) for mapper, ds in zip(mappers, dss_test)]
+
+        # assign .sa.subjects to those datasets
+        for i, ds in enumerate(dss_test_aligned):  ds.sa["subjects"] = [i]
+
+        dss_test_bc = []
+        for ds in dss_test_aligned:
+            if overlapping_windows:
+                startpoints = range(len(ds) - window_size)
+            else:
+                raise NotImplementedError
+            ds_ = BoxcarMapper(startpoints, window_size).forward(ds)
+            ds_.sa['startpoints'] = startpoints
+            # reassign subjects so they are not arrays
+            def assign_unique(ds, sa):
+                ds.sa[sa] = [np.asscalar(np.unique(x)) for x in ds.sa[sa].value]
+            for saname in ['subjects']:
+                assign_unique(ds_, saname)
+
+            fm = FlattenMapper(); fm.train(ds_)
+            dss_test_bc.append(ds_.get_mapped(fm))
+
+        ds_test = vstack(dss_test_bc)
+        # Perform classification across subjects comparing against mean
+        # spatio-temporal pattern of other subjects
+        errors_across_subjects = []
+        for ds_test_part in part2.generate(ds_test):
+            ds_train_, ds_test_ = list(Splitter("partitions").generate(ds_test_part))
+            ds_train_ = mean_group_sample(['startpoints'])(ds_train_)
+            assert(ds_train_.shape == ds_test_.shape)
+
+            if distance == 'correlation':
+                # TODO: redo
+                dist = 1 - np.corrcoef(ds_train_, ds_test_)[len(ds_test_):, :len(ds_test_)]
+            else:
+                raise NotImplementedError
+
+            if overlapping_windows:
+                dist = zero_out_offdiag(dist, window_size)
+
+            # For Yarik it feels that we need axis=1 here!! TODO
+            winners = np.argmin(dist, axis=int(clf_direction_correct_way))
+            #print winners
```

Comment:
You can also remove the variable `clf_direction_correct_way`, and make it always do axis=1.